tests/python_functional/conftest.py

```diff
@@ -65,6 +71,12 @@ def get_current_date():
 
 # Pytest Hooks
 def pytest_collection_modifyitems(config, items):
+    if config.getoption("--skip-tests-with-external-deps"):
```

Comment:
Sorry about that, we have found that pytest has an internal support to run explicit markers, for example:
```console
python -m pytest ... -m 'external_dependency'
```
- so with this I would remove this '--skip-tests-with-external-deps' option, and instead of this user can use `-m 'some-marker'` option

Other thing:
Pytest can discover all markers inside tests with
```config
python -m pytest --markers
```
- so with this we can add more specific markers like 'snmp', If we will have a lot of markers user can discover it, and add them separately (customized):
```
python -m pytest ... -m 'snmp and other-module and not some-other-modul'
```
- so for now I would rename our 'external_dependency' to more concrete: 'snmp'

Reply:
done. now you can control the running of snmp tests by the -m 'snmp' parameter